### DISCUSSION TITLE
How to set hsplit/vsplit window sizes programmatically?

### DISCUSSION THREAD

**Discussion by carlosabalde** [kovidgoyal/kitty#5343] [Q&A] [answered]
Hi,

I'm trying to use Kitty to programmatically (1) create a new OS window using `launch`; then (2) create a slightly complex mosaic of windows also using the `splits` layout and `launch` + `goto-layout` + `focus-window`; and then (3) execute different actions in each window using `send-text`. Example:
  - 4 rows (each one using 15% of the total height) + 1 row (using 25% height) + 1 row using remaining height (i.e. 30%). 
  - For first four rows, each one containing 2 columns (each one using 50% of the total width).
  - For fifth row, 3 columns (each one using 20%, 20% and 60% of total width).
  - For sixth row, just 1 column.

All seems doable, nice and clean using remote control feature, except the definition of sizes for each window in the mosaic. Basically I'd expect to be able to use `hsplit` and `vsplit` to specify a percentage, but that doesn't seem to be possible. I've also considered using `resize-window`, but the only option is `--increment` which seems to be a non-go for this. I've been using Kitty just for a few weeks, so I'm not sure if I'm missing anything obvious. Any hints would be highly appreciated :)

**Comment by carlosabalde** [comment]
Based on https://github.com/kovidgoyal/kitty/issues/3620, it seems what I want to do is simply not doable at the moment. It seems I should use `resize-window` and implement some logic to calculate values for `--increment`. In order to do that, I'll need to be able to fetch with and height of windows. I've seen https://sw.kovidgoyal.net/kitty/graphics-protocol/#getting-the-window-size, but how to do that using the remote control feature?

**Comment by kovidgoyal** [comment, marked as answer]
kitty @ ls gives you all the information you need.

**Reply by carlosabalde** [reply]
Thanks for the hint!
